packages/drafts-realm/cycling-mileage-log.gts

```diff
@@ -0,0 +1,148 @@
+import {
+  contains,
+  containsMany,
+  CardDef,
+  FieldDef,
+  StringField,
+  field,
+} from 'https://cardstack.com/base/card-api';
+import NumberField from 'https://cardstack.com/base/number';
+import { Component } from 'https://cardstack.com/base/card-api';
+import Date from 'https://cardstack.com/base/date';
+// @ts-ignore
+import {
+  Chart,
+  registerables,
+  // @ts-ignore
+} from 'https://cdn.jsdelivr.net/npm/chart.js@4.4.2/+esm';
+import Modifier from 'ember-modifier';
+
+type ProgressChartModifierSignature = {
+  Args: {
+    Named: {
+      target?: number;
+      total?: number;
+    };
+  };
+  Element: HTMLElement;
+};
+
+class ProgressChartModifier extends Modifier<ProgressChartModifierSignature> {
+  modify(
+    element: HTMLElement,
+    _positional: [],
+    { target, total }: ProgressChartModifierSignature['Args']['Named'],
+  ) {
+    total = total ?? 0;
+    target = target ?? 0;
+    Chart.register(...registerables);
+    new Chart(element, {
+      type: 'doughnut',
+      data: {
+        labels: ['progress (KM)', 'remaining (KM)'],
+        datasets: [
+          {
+            label: ['Cycling Mileage Log'],
+            data: [total, target - total],
+            backgroundColor: ['rgb(255, 99, 132)', 'rgb(54, 162, 235)'],
+            hoverOffset: 4,
+          },
+        ],
+      },
+    });
+  }
+}
+
+class CyclingLogEntry extends FieldDef {
+  static displayName = 'Cycling Log Entry';
+
+  @field date = contains(Date);
+  @field origin = contains(StringField);
+  @field destination = contains(StringField);
+  @field distance = contains(NumberField);
+
+  static embedded = class Embedded extends Component<typeof this> {
+    <template>
+      <div class='entry'>
+        <span>Date: <@fields.date /></span>
+        <span>Routes: <@fields.origin /> - <@fields.destination /></span>
+        <span>Distance: <@fields.distance /> KM</span>
+      </div>
+      <style>
+        .entry {
+          display: flex;
+          flex-direction: column;
+          padding: var(--boxel-sp);
+          border: var(--boxel-border);
+        }
+      </style>
+    </template>
+  };
+}
+
+export class CyclingMileageLog extends CardDef {
+  static displayName = 'Cycling Mileage Log';
+
+  @field target = contains(NumberField);
+  @field total = contains(NumberField, {
+    computeVia: function (this: CyclingMileageLog) {
+      let sum = 0;
+      for (let entry of this.entries) {
+        sum += entry.distance;
+      }
+      return sum;
+    },
+  });
+  @field successPercentage = contains(NumberField, {
+    computeVia: function (this: CyclingMileageLog) {
+      return this.total / this.target;
```

Comment:
Round this number, otherwise many decimals are shown

<img width="291" alt="image" src="https://github.com/cardstack/boxel/assets/273660/2a07343a-2c40-47d3-a033-6c5b6ffd2e66">